CONTRIBUTING.md

```diff
@@ -8,24 +8,59 @@ After following the steps in the [Contribution Guide](http://docs.mattermost.com
 
 The one exception may be around release time, where the review process may take longer as the team focuses on our [release process](https://docs.mattermost.com/process/release-process.html). 
 
-#### `Stage 1: PM Review`
+#### PR submitted
+Dev applies labels and alerts PMs that there is a PR awaiting review by posting in the [PM/Docs PR Review channel](https://pre-release.mattermost.com/core/channels/pmdocs-pr-review-pub)
+ - `Awaiting PR`
+  - Applied if the PR is awaiting for another to be merged. For exampl,e when a client PR is awaiting a server PR to be merged first.
+  - PR is on hold until it's no longer blocked, then dev removes the `Awaiting PR` label
+ - `1: PM Review`
+  - Applied if PR has UI changes or functionality that PMs can test on spinmints
+ - `Setup Test Server`
+  - Applied if the PR is queued for PM review for testing
+ - `Work in Progress`
+  - Applied if the PR is unfinished
+  - PR is not reviewed until the label is removed
+
+#### Stage 1: PM Review
 
 A Product Manager will review the pull request to make sure it:
 
-1. Fits with our product roadmap
-2. Works as expected
-3. Meets UX guidelines
+ - Fits with our product roadmap
+ - Works as expected
+ - Meets UX guidelines
 
-This step is sometimes skipped for bugs or small improvements with a ticket, but always happens for new features or pull requests without a related ticket. 
+This step is sometimes skipped for bugs or small improvements with a ticket, but always happens for new features or pull requests without a related ticket.
 
 The Product Manager may come back with some bugs or UI improvements to fix before the pull request moves on to the next stage.
 
-#### `Stage 2: Dev Review`
+- PM applies milestone:
+ - Set for next release if the PM thinks there is enough time for the PR to be merged and sufficiently tested on `master` before code complete.
+ - Set for a future release if PR is too large to test and merge prior to code complete date
+   - PM responds to submitter letting them know that PR may have a delay in review due to the release cycle and can be taken in `master` after the release branch is cut 
+
+- PM tests on spinmint:
```

Comment:
Does everybody know what spinmint means? I did a google search and got nothing related.